### PR TITLE
Remove internal ConfigQueryResult type

### PR DESCRIPTION
**Pull request by Choc13**
* Removed `IConfigQueryResult` and `ConfigQueryResult`
* Cleaned up ConsulConfigurationProviderTests

### CODE REVIEW

File: src/Winton.Extensions.Configuration.Consul/ConsulConfigurationClient.cs [hunk 1]
```diff
@@ -16,21 +16,21 @@ internal sealed class ConsulConfigurationClient : IConsulConfigurationClient
         private readonly IConsulClientFactory _consulClientFactory;
         private readonly object _lastIndexLock = new object();
         private readonly IConsulConfigurationSource _source;
+        private ulong _lastIndex;
 
         private ConfigurationReloadToken _reloadToken = new ConfigurationReloadToken();
-        private ulong _lastIndex;
```
Comment (jhickson) [inline src/Winton.Extensions.Configuration.Consul/ConsulConfigurationClient.cs]: Any reason for this moving? It was better where it originally was, wasn't it?

File: src/Winton.Extensions.Configuration.Consul/ConsulConfigurationClient.cs [hunk 2]
```diff
@@ -39,23 +39,37 @@ public IChangeToken Watch(Action<ConsulWatchExceptionContext> onException)
...
+                        false);
                 switch (result.StatusCode)
                 {
                     case HttpStatusCode.OK:
                     case HttpStatusCode.NotFound:
                         return result;
                     default:
-                        throw new Exception($"Error loading configuration from consul. Status code: {result.StatusCode}.");
+                        throw new Exception(
+                            $"Error loading configuration from consul. Status code: {result.StatusCode}.");
```
Comment (jhickson) [inline src/Winton.Extensions.Configuration.Consul/ConsulConfigurationClient.cs]: Any point to this change?

File: src/Winton.Extensions.Configuration.Consul/ConsulConfigurationClient.cs [hunk 3]
```diff
@@ -39,23 +39,37 @@ public IChangeToken Watch(Action<ConsulWatchExceptionContext> onException)
...
-        private async Task<QueryResult<KVPair>> GetKVPair(QueryOptions queryOptions = null)
+        private async Task<QueryResult<KVPair>> GetKvPair(QueryOptions queryOptions = null)
         {
             using (IConsulClient consulClient = _consulClientFactory.Create())
             {
-                QueryResult<KVPair> result = await consulClient.KV.Get(_source.Key, queryOptions, _source.CancellationToken)
-                    .ConfigureAwait(false);
+                QueryResult<KVPair> result =
+                    await consulClient.KV.Get(_source.Key, queryOptions, _source.CancellationToken).ConfigureAwait(
+                        false);
```
Comment (jhickson) [inline src/Winton.Extensions.Configuration.Consul/ConsulConfigurationClient.cs]: If there's to be a new line could the `.ConfigureAwait(` move too? Looks odd with just `false);`.

File: src/Winton.Extensions.Configuration.Consul/ConsulConfigurationClient.cs [hunk 4]
```diff
@@ -64,7 +78,9 @@ public IChangeToken Watch(Action<ConsulWatchExceptionContext> onException)
                 {
                     if (await HasValueChanged().ConfigureAwait(false))
                     {
-                        var previousToken = Interlocked.Exchange(ref _reloadToken, new ConfigurationReloadToken());
+                        ConfigurationReloadToken previousToken = Interlocked.Exchange(
+                            ref _reloadToken,
+                            new ConfigurationReloadToken());
```
Comment (jhickson) [inline src/Winton.Extensions.Configuration.Consul/ConsulConfigurationClient.cs]: Revert to original using `var`?

File: src/Winton.Extensions.Configuration.Consul/ConsulConfigurationProvider.cs [hunk 5]
```diff
@@ -15,7 +17,9 @@ internal sealed class ConsulConfigurationProvider : ConfigurationProvider
         private readonly IConsulConfigurationClient _consulConfigClient;
         private readonly IConsulConfigurationSource _source;
 
-        public ConsulConfigurationProvider(IConsulConfigurationSource source, IConsulConfigurationClient consulConfigClient)
+        public ConsulConfigurationProvider(
+            IConsulConfigurationSource source,
+            IConsulConfigurationClient consulConfigClient)
```
Comment (jhickson) [inline src/Winton.Extensions.Configuration.Consul/ConsulConfigurationProvider.cs]: Revert to original?

File: src/Winton.Extensions.Configuration.Consul/ConsulConfigurationProvider.cs [hunk 6]
```diff
@@ -41,65 +45,57 @@ public override void Load()
...
+            }
+        }
+
         private async Task DoLoad(bool reloading)
         {
             try
             {
-                IConfigQueryResult configQueryResult = await _consulConfigClient.GetConfig().ConfigureAwait(false);
-                if (!configQueryResult.Exists && !_source.Optional)
+                QueryResult<KVPair> queryResult = await _consulConfigClient.GetConfig().ConfigureAwait(false);
```
Comment (jhickson) [inline src/Winton.Extensions.Configuration.Consul/ConsulConfigurationProvider.cs]: Use `var`.

File: test/Winton.Extensions.Configuration.Consul.Test/ConsulConfigurationClientTests.cs [hunk 7]
```diff
@@ -71,7 +71,7 @@ public void TearDown()
             {
                 StatusCode = statusCode
             };
-            IConfigQueryResult configQueryResult = await SimulateGet(result);
+            QueryResult<KVPair> configQueryResult = await SimulateGet(result);
```
Comment (jhickson) [inline test/Winton.Extensions.Configuration.Consul.Test/ConsulConfigurationClientTests.cs]: Use `var`.

**Comment by Choc13** [comment]
@jhickson it seems all your comments are about stylistic things. They were all changes that R# code cleanup made to the project. I was using the DotSettings that we use in our team internally. For the points regarding using `var` I would prefer to use the explicit type when it is not clear from the RHS of the expression. As a broader point, I can revert the whitespace changes and do another PR afterwards where I run R# code cleanup over the entire project. On that note, do we have a DotSettings file that we use for our open source projects.

**Comment by jhickson** [comment]
@Choc13 I don't believe we have a DotSettings file for use with our open-source projects. Perhaps the DotSettings file you're using should be added to the repo and, at the same time, there should be a project-wide code clean-up. That should probably be a separate PR though usually I'm fine with small amounts of clean-up just being isolated into their own commit - that way you can easily review the functional changes without the clutter.

As for `var`, I prefer to always use it as far as possible. There are lots of situations other than variable assignment where the type is not obvious e.g.

```
Foo(Bar())
```

If you wanted to remove ambiguity here you'd need

```
SomeType s = Bar();
Foo(s);
```

but people are usually perfectly happy with the first form and so, by extension, should also be happy with `var` everywhere. Using `var` everywhere tends to encourage more functional code too.

Anyway, it's up to you. I think my main issue is with the way some of the line breaks are applied - they just look messy.

**Comment by Choc13** [comment]
@jhickson I've opened #29 to do a solution wide code cleanup. R# recently got a new [formatting engine](https://blog.jetbrains.com/dotnet/2017/11/23/code-formatting-engine-updates-resharper-rider/) and I agree I do find some of it's line breaks messy, but it quickly becomes a futile task fighting automated tooling like that. We have enabled a flag which reverts to the old formatting engine in the DotSettings file, but it doesn't seem to be exactly equivalent. Anyway, I have for the most part tried to help R# out a bit in places where it has made some really odd formatting decisions. 

That's a good point about `var`, and yes I am perfectly fine with `Foo(Bar())` and I agree that encouraging a more functional style is a positive thing. I guess in the few cases where an intermediate variable is required then I do prefer to see the type explicitly as I find it helps to make the code more readable when the return type of an expression is stated (particularly when there are chained method calls, such as with linq). Obviously this is largely a preference thing, so I have opted to stick with the R# settings as they currently are.

File: test/Website/Program.cs [hunk 8]
```diff
@@ -7,7 +7,7 @@ public class Program
     {
         public static void Main(string[] args)
         {
-            var host = new WebHostBuilder()
+            IWebHost host = new WebHostBuilder()
```
Comment (Choc13) [inline test/Website/Program.cs]: Sorry about this, I don't know why R# didn't catch this file in the previous code cleanup

File: test/Website/Startup.cs [hunk 9]
```diff
@@ -11,59 +11,47 @@ namespace Winton.Extensions.Configuration.Consul.Website
 {
     public class Startup
     {
-        private readonly CancellationTokenSource _consulConfigCancellationTokenSource = new CancellationTokenSource();
         private readonly IConfigurationRoot _configuration;
+        private readonly CancellationTokenSource _consulConfigCancellationTokenSource = new CancellationTokenSource();
```
Comment (Choc13) [inline test/Website/Startup.cs]: Same in here, not sure why R# missed it last time.

**Pull request merged**